src/main/java/com/getindata/connectors/http/internal/table/lookup/HttpLookupTableSource.java

```diff
@@ -94,21 +100,41 @@ public LookupRuntimeProvider getLookupRuntimeProvider(LookupContext lookupContex
         PollingClientFactory<RowData> pollingClientFactory =
             createPollingClientFactory(lookupQueryCreator, lookupConfig);
 
-        HttpTableLookupFunction dataLookupFunction =
-            new HttpTableLookupFunction(
-                pollingClientFactory,
-                responseSchemaDecoder,
-                lookupRow,
-                lookupConfig
-            );
+        // In line with the JDBC implementation and current requirements, we are only
+        // supporting Partial Caching for synchronous operations.
+        return getLookupRuntimeProvider(lookupRow, responseSchemaDecoder, pollingClientFactory);
+    }
 
+    protected LookupRuntimeProvider getLookupRuntimeProvider(LookupRow lookupRow,
+        DeserializationSchema<RowData> responseSchemaDecoder,
+        PollingClientFactory<RowData> pollingClientFactory) {
         if (lookupConfig.isUseAsync()) {
+            HttpTableLookupFunction dataLookupFunction =
+                    new HttpTableLookupFunction(
+                            pollingClientFactory,
+                            responseSchemaDecoder,
+                            lookupRow,
+                            lookupConfig
+                    );
             log.info("Using Async version of HttpLookupTable.");
             return AsyncTableFunctionProvider.of(
                 new AsyncHttpTableLookupFunction(dataLookupFunction));
         } else {
-            log.info("Using blocking version of HttpLookupTable.");
-            return TableFunctionProvider.of(dataLookupFunction);
+            CachingHttpTableLookupFunction dataLookupFunction =
+                    new CachingHttpTableLookupFunction(
+                            pollingClientFactory,
+                            responseSchemaDecoder,
+                            lookupRow,
+                            lookupConfig,
+                            cache
+                    );
+            if (cache != null) {
```

Comment:
I'm not super happy that we can pass `null` as cache to `HttpLookupSource`. Plus if `cache` is null you still create an instance of `CachingHttpTableLookupFunction`. 

I'm also thinking whether `cache` should be passed to `HttpLookupTableSource`
We have `HttpLookupConfig` that contains entire config (via `ReadableConfig`) that is already passed to HttpLookupTableSource. You could move `getLookupCache` from `HttpLookupTableSourceFactory` to `HttpLookupTableSource` and create cache there.

Going step futher, you could have two implementations for `HttpLookupTableSource` -> cached and noCache, build a factory class for it and move `getLookupCache` to that factory